build/Dockerfile

```diff
@@ -12,57 +12,58 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM registry.access.redhat.com/ubi9/python-311 as wheel
 
-ARG WHEEL_VERSION=""
-USER root
-RUN --mount=type=cache,target=/root/.cache/pip \
-    python -m pip install --upgrade pip && \
-    python -m pip install build
-COPY tuning tuning
-COPY .git .git
-COPY pyproject.toml pyproject.toml
-# build wheel if wheel version is empty else download the wheel from PyPi
-RUN if [[ -z "${WHEEL_VERSION}" ]]; \
-    then python -m build --wheel --outdir /tmp; \
-    else pip download fms-hf-tuning==${WHEEL_VERSION} --dest /tmp --only-binary=:all: --no-deps; \
-    fi && \
-    ls /tmp/*.whl >/tmp/bdist_name
-
-
-FROM registry.access.redhat.com/ubi9/ubi AS release
-
-ARG CUDA_VERSION=11.8.0
+## Global Args #################################################################
+ARG BASE_UBI_IMAGE_TAG=latest
 ARG USER=tuning
 ARG USER_UID=1000
-ARG SET_NUM_PROCESSES_TO_NUM_GPUS=True
+ARG PYTHON_VERSION=3.11
+ARG WHEEL_VERSION=""
+
+## Base Layer ##################################################################
+FROM registry.access.redhat.com/ubi9/ubi:${BASE_UBI_IMAGE_TAG} as base
 
-USER root
+ARG PYTHON_VERSION
+ARG USER
+ARG USER_UID
 
 RUN dnf remove -y --disableplugin=subscription-manager \
         subscription-manager \
-        # we install newer version of requests via pip
-        python3.11-requests \
-    && dnf install -y make \
-        # to help with debugging
-        procps \
+    && dnf install -y python${PYTHON_VERSION} procps \
+    && ln -s /usr/bin/python${PYTHON_VERSION} /bin/python \
+    && python -m ensurepip --upgrade \
+    && dnf update -y \
     && dnf clean all
 
 ENV LANG=C.UTF-8 \
     LC_ALL=C.UTF-8
 
-ENV CUDA_VERSION=$CUDA_VERSION \
-    NV_CUDA_LIB_VERSION=11.8.0-1 \
+RUN useradd -u $USER_UID ${USER} -m -g 0 --system && \
+    chmod g+rx /home/${USER}
+
+## Used as base of the Release stage to removed unrelated the packages and CVEs
+FROM base as release-base
+
+# Removes the python3.9 code to eliminate possible CVEs.  Also removes dnf
+RUN rpm -e $(dnf repoquery python3-* -q --installed) dnf python3 yum crypto-policies-scripts
```

Comment:
@jbusche can you verify the change still works as intended ?

Reply:
Hey @tedhtchang, yes I'm not seeing any fixable Twistlock CVEs with the above.